### PR TITLE
Ignore Duplicate based on ExecuteType (Fixes #194)

### LINKED ISSUE #194: `Connection Open()`/`Connection OpenAsync()` Marked as Duplicate
I don't necessarily think it's a problem when a single HTTP request happens to make multiple calls to  Open, and so this creates a lot of false-positives in the UI.

Since this is new behavior in v4, I think there are two ways to handle this:

1. Always ignore ExecuteType === "Open"/OpenAsync" when detecting duplicates in the UI.
2. A new setting (`IgnoreDuplicateOpenConnection`?) to conditionally control 1.

Whichever approach we like, I'll throw together a pull request.

### PR DESCRIPTION
**Pull request by jdaigle**
Introducing a new "CustomTimingExecuteTypesToIgnoreAsDuplicate" setting to
explicitly ignore duplicate custom timings based on its execute type.

By default it will ignore "Open" and "OpenAsync".

### CODE REVIEW

File: src/MiniProfiler.Shared/Internal/MiniProfilerExtensions.cs [hunk 1]
```diff
@@ -34,7 +35,8 @@ public static class MiniProfilerExtensions
             bool? showTimeWithChildren = null,
             int? maxTracesToShow = null,
             bool? showControls = null,
-            bool? startHidden = null)
+            bool? startHidden = null,
+            HashSet<string> customTimingExecuteTypesToIgnoreAsDuplicate = null)
```
Comment (NickCraver) [inline src/MiniProfiler.Shared/Internal/MiniProfilerExtensions.cs]: IMO accepting this here is overkill - I'd just make it a setting

File: src/MiniProfiler.Shared/MiniProfiler.Settings.cs [hunk 2]
```diff
@@ -148,6 +149,15 @@ public static class Settings
             public static bool ShowControls { get; set; } = false;
 
             /// <summary>
+            /// Custom timing ExecuteTypes to ignore as duplicates in the UI.
+            /// </summary>
+            public static HashSet<string> CustomTimingExecuteTypesToIgnoreAsDuplicate { get; } = new HashSet<string>()
```
Comment (NickCraver) [inline src/MiniProfiler.Shared/MiniProfiler.Settings.cs]: `CustomTimingExecuteTypesToIgnoreAsDuplicate` is a bit long, how about `IgnoredDuplicateExecuteTypes`?

File: src/MiniProfiler.Shared/MiniProfiler.Settings.cs [hunk 3]
```diff
@@ -148,6 +149,15 @@ public static class Settings
...
+            {
+                nameof(DbConnection.Open),
```
Comment (NickCraver) [inline src/MiniProfiler.Shared/MiniProfiler.Settings.cs]: Don't use `System.Data` here, instead use `nameof(ProfiledDbConnection.Open)` (what's actually used) - also makes "Find All References", etc. work :)

File: src/MiniProfiler.Shared/ui/includes.js [hunk 4]
```diff
@@ -216,7 +216,7 @@ var MiniProfiler = (function () {
                     customTiming.ParentTimingId = timing.Id;
                     customStat.Duration += customTiming.DurationMilliseconds;
                     customStat.Count++;
-                    if (customTiming.CommandString && duplicates[customTiming.CommandString]) {
+                    if (customTiming.CommandString && duplicates[customTiming.CommandString] && !ignoreDuplicateCustomTiming(customTiming)) {
```
Comment (NickCraver) [inline src/MiniProfiler.Shared/ui/includes.js]: I'd change this to be simpler: remove this change and instead modify the `else` below to `else if (!ignoreDuplicateCustomTiming(customTiming))`, so it never gets added to `duplicates` below, that is less likely to get tripped on for uses of `duplicates` later.

**Comment by jdaigle** [comment]
Implemented suggested changes and tested.

**Comment by NickCraver** [comment]
Looks great, thanks @jdaigle!

**Pull request merged**